database/rrdset.c

```diff
@@ -519,18 +520,138 @@ RRDSET *rrdset_create_custom(
+        if (unlikely((st->name && !strcmp(st->name, new_name)) || !st->name)) {
+            mark_rebuild |= META_CHART_UPDATED;
+            rrdset_set_name(st, new_name);
+        }
+
+        if (unlikely(st->priority != priority)) {
+            st->priority = priority;
+            mark_rebuild |= META_CHART_UPDATED;
+        }
+        if (unlikely(st->rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE && st->update_every != update_every)) {
+            st->update_every = update_every;
+            mark_rebuild |= META_CHART_UPDATED;
+        }
+
+        if (plugin && st->plugin_name) {
+            if (unlikely(strcmp(plugin, st->plugin_name))) {
+                old_plugin = st->plugin_name;
+                st->plugin_name = strdupz(plugin);
+                mark_rebuild |= META_PLUGIN_UPDATED;
+            }
+        } else {
+            if (plugin != st->plugin_name) { // one is NULL?
+                old_plugin = st->plugin_name;
+                st->plugin_name = plugin ? strdupz(plugin) : NULL;
+                mark_rebuild |= META_PLUGIN_UPDATED;
+            }
+        }
 
+        if (module && st->module_name) {
+            if (unlikely(strcmp(module, st->module_name))) {
+                old_module = st->module_name;
+                st->module_name = strdupz(module);
+                mark_rebuild |= META_MODULE_UPDATED;
+            }
+        } else {
+            if (module != st->module_name) {
+                if (st->module_name && *st->module_name) {
+                    old_module = st->module_name;
+                    st->module_name = module ? strdupz(module) : NULL;
+                    mark_rebuild |= META_MODULE_UPDATED;
+                }
+            }
+        }
+
+        char *new_title = title ? strdupz(title) : NULL;
+        if (new_title) {
+            json_fix_string(new_title);
+            if (unlikely(strcmp(st->title, new_title))) {
+                old_title = st->title;
+                st->title = new_title;
+                mark_rebuild |= META_CHART_UPDATED;
+            } else
+                freez(new_title);
+        }
+
+        RRDSET_TYPE new_chart_type =
+            rrdset_type_id(config_get(st->config_section, "chart type", rrdset_type_name(chart_type)));
+        if (st->chart_type != new_chart_type) {
+            st->chart_type = new_chart_type;
+            mark_rebuild |= META_CHART_UPDATED;
+        }
+
+        char *new_family = family ? strdup(family): NULL;
+        if (new_family) {
+            json_fix_string(new_family);
+            if (unlikely((strcmp(st->family, new_family)))) {
+                old_family = st->family;
+                rrdfamily_free(host, st->rrdfamily);
+                st->family = new_family;
+                st->rrdfamily = rrdfamily_create(host, st->family);
+                mark_rebuild |= META_CHART_UPDATED;
+            } else
+                freez(new_family);
+        }
+
+        char *new_context = context ? strdup(context) : NULL;
```

Comment:
Same as above.

Reply:
I think that it needs a `strdupz` here as the `json_fix_string` rewrites the input.